server/src/test/java/org/elasticsearch/search/DefaultSearchContextTests.java

```diff
@@ -475,6 +476,30 @@ public void testNewIdLoaderWithTsdb() throws Exception {
         }
     }
 
+    public void testNewIdLoaderWithTsdbAndRoutingPathMatch() throws Exception {
```

Comment:
This test is good. What I wonder now is whether `TsIdLoader` correct synthesises ids when a routing path with wildcards is used. I don't expect this to be the case. I don't see a test and `TsIdLoader` doesn't load routing fields based on wildcard. If this is the case, then this is an existing bug. No need to fix it here, but should be addressed in a follow up change.

Reply:
I think the changes in this PR (modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java) lead to using routing path with wildcards. I think the added yaml tests exercise it, and we can be more explicit for mappings without the pass-through object.